### ISSUE TITLE
lint: inconsistent fenced code block `-`line indentation

### ISSUE THREAD

**Issue by jorgeorpinel** [iterative/dvc.org#554]
A problem after implementing #372 is that some lines inside md fenced code blocks starting with `-` get indented for unknown reasons by Prettier.

E.g. by finding the regex `^- cache` in all files we can see that static/docs/commands-reference/commit.md has the unindented lines:

![image](https://user-images.githubusercontent.com/1477535/62908810-84cb0e00-bd3f-11e9-8b1f-78273cb4bd5a.png)
> Similar cases in static/docs/get-started/connect-code-and-data.md and static/docs/tutorial/define-ml-pipeline.md

However static/docs/commands-reference/add.md has them indented:

![image](https://user-images.githubusercontent.com/1477535/62908946-176bad00-bd40-11e9-8c54-76b060851190.png)

I don't see a clear pattern explaining this apparently erratic behavior but it's all over the docs.

Again, this i work of Prettier which has been run on all md files in #512. None of the DVC-file samples had indentation before `-` lines originally. We decided that the indentation was OK in https://github.com/iterative/dvc.org/issues/512#issuecomment-518878493 but the fact that it indents some and unindents others troubles me a little. Perhaps we should use `<!-- prettier-ignore -->`s before all YAML code blocks after all?

All these code samples are still valid YAML though so please just close this issue if the decision is to live with it and let `prettier` do whatever it wants 😋 

---

Originally reported partially as [lint: don't indent fenced code block contents](https://github.com/iterative/dvc.org/issues/512)

**Comment by shcheklein** [comment]
I would actually create a ticket on the Prettier's repo to be honest. For us it does not look very important to include special instructions.

**Comment by shcheklein** [comment]
Closing this for now :)

**Comment by jorgeorpinel** [comment]
Good call, I started opening an issue at https://github.com/prettier/prettier/issues/new?template=formatting.md but in the process I discovered why this is happening!

The blocks that don't get indented have repeated keys inside the YAML blocks so **actually they're NOT valid YAML**. Prettier tries to run the YAML linter inside, but simply fails to parse them with error message `SyntaxError: Map keys must be unique;` (See [this playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdKADLBPAhgWwBt0x8ATJAAgAdsYALaSgZwCcwB6GV3ASygB0tSmVwxcHAGZwxAV1ZxmlfBDJxCQgNbEoa6syToAtMrIBWKmQAMZuFYBMVgIy4AzJIBGAFgDsYABw+-mrSuABsZh5k-rZeZOiUNGL0VGyc3HyCtMamFpQePm5mPh64PpLhBfZhcGFOZmRkYfYerl5OAJxW3gJkvKwJSQyWYhKhMPKKg+R5Hh1OHvYtktFkHXBmuJJOrnD+kl4x9rv+DmZW6BCyMAY5YLhg9HBU3LJw0+ZUPjYdHdthYTAjg6wXKknu8zcYQ6ZX8His8SgiXwcG4vDAVAqhGY7yRQxSylU6i0OkS1DgrGYvGYMExuGxuIA7n1WFQBOgsBh0CAADQgCDUGC8aDMZCgXCsVgQRkABQlCFFKFwADcILwyLyQB4eGBNKiAMrUB78ADmyFecD59BgRAA6vRePBmEawHB9QrHbxlY7sMhwMxRXz+DjWDAZTwTfhcMgsTi+QArZgADwAQjq9TB9QQ4AAZfhwGP0uMgRNJ-WmwhwACKsgg8ELDL5RspFL9UdYmjI0qgmuorH4MFt6oYyFOTalONtPGofr7igpyoLfIAjrX4OGBYqQLhmEYoHA4GoNXyFKv+nBw7hI9GkLHLSAcfheObWG8+VSoCbKzW6wXb0X73EDwhzIEckHsPkMl4QhTQAYQgfAoz9KBoCXEBZBxAAVXAPEVO8AF98KAA)).

Should we be worried that some of our DVC-file aren't technically valid YAML @shcheklein ? More of a Q for the core repo but asking here anyway.

**Comment by shcheklein** [comment]
there is a bug in docs - in the first example, third md5 from the top should not be indented at all. It's a top-level md5.

**Comment by shcheklein** [comment]
reopening it to check the docs and linting for this bug

**Comment by jorgeorpinel** [comment]
Hmm I see. OK I fixed that in 5f13d5c but what about the other 2 examples I mentioned?

Here's a [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEADdBPAhgWwDYA6UYOAJkgAQAOGMAFtBQM4BOYA9FS3FVtwHQ0KpLDCzsRY-gA98RUjyZIiAWgpkArJQBGAFgAcABgCMYffqz6AzADMr2mxsMAmXWGMBOLB93OPHqywDIl56SlYOLh4+OEEMVXVSLQosK0NdLBttDwA2QzysLFIPEwB2HOcc610NOGNDKw0NENE6SklxDpk5KE1KMEDSq3qPbTgc930c7NJS5zhS411tMDgRUlJtIggAVxglBLAsMDo4ShgWHbgiPooqqxybUo9Suw9XLN1HsBtSaY8bDlCjldMYcvxSABLFg3OAXSFgSg2LB4JjXKChNrCUTiKK8bikIgAdyhLEo-CI6FQRBAABoQBAqDBIdAmMhQHwWBAiQAFGJQNkoLAANwgkNIdJA2hYxwA1nCAMq8MCQqAAc2QFyu9LoMHwAHU6JD4ExlXAFQgmMbIcLjRhkOAmGz6aq0SwYDyZWqcFhkMjUXB6QArJjSABCMrA8pgCtwcAAMqq4H6UWjg6GFaq1Xg4ABFHYQeApgP0-Fuh0+liy0jcqCSriqmD68X0ZBGUtctH6mVUB1RN3C5P0gCOBfgnsZgpAWCYKigcDWa0l3FH0Lgnqw3t9SH9aZAaJwkM1l0D+6zOfzheTO9Tp7E2mbpFbSGc9IuWEheCzAGEIDgfQ6UDQEOIA7GiAAqWDaIKu5wAAvnBQA) for static/docs/get-started/connect-code-and-data.md. In this case it complaints that `SyntaxError: Document is not valid YAML (bad indentation?) (5:1)`

And in this [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEADdATAbmAOlAEgAIwBDGIjc0gegAUIBnGRgOgA8BbAG1YC8AlgAdW2PFE4YArEiIB2KVICccAEYAWKarVy4SgAwAzAIxKATHAAcAZgBs2y-utnSt9fggBXFknwBaElIwAAs4WRgAJ084fCIiSRkiODBjDDMlaylDS2NjVTk5MDB1JVUnPKVKsFUTOViiIXJg2QZmNi5eQSF8fGIMTyI-RmCiURwaMhC4GmSaACpY9WMAWVGxCaDQmbAaVPTM7Nz8wuLS8tVKpWra-HRUfBAAGhAIIRgBaEZkUFIIiIgAO50X4IL4oUhYCACDBPECqCJBADWcBgAGVGmABFAAObISLRZ7BGA8ADqwQE8EYGLgqNBFIEWApAE9kOBGF9nljGHAIjA6AjsZxSMhDKRuNzngArRjsABCCLAyLRpE4cAAMli4CKxRKQNL2KisdjuHAAIqeCDwbXiuDPRoRbkRVlCiKIjCAqCwoQRLEwEnQmDBZCOO3-bkkhFCVneuCOrBa54ARwt8H5rzBIFIjD8UDgcAw+dhETgyYExf5pEFwqQoptz25nAEeKitpAjCNJvNlq1NZ1rZgpFU-owgeQZmekVIAm4RoAwhBOELWVBoAmQJ5uQAVQdg2vcgC++6AA) for static/docs/tutorial/define-ml-pipeline.md quite simply since the block langauge is ```dvc (unknown to Prettier) it ignores it. I can fix that by making sure all YAML samples are in '''yaml language blocks.

**Comment by shcheklein** [comment]
The first one should be like this:

```yaml
cmd: python src/prepare.py data/data.xml
deps:
- md5: b4801c88a83f3bf5024c19a942993a48
  path: src/prepare.py
- md5: a304afb96060aad90176268345e10355
  path: data/data.xml
md5: c3a73109be6c186b9d72e714bcedaddb
outs:
- cache: true
  md5: 6836f797f3924fb46fcfd6b9f6aa6416.dir
  metric: false
  path: data/prepared
wdir: .
```
or like this:

```yaml
cmd: python src/prepare.py data/data.xml
deps:
  - md5: b4801c88a83f3bf5024c19a942993a48
    path: src/prepare.py
  - md5: a304afb96060aad90176268345e10355
    path: data/data.xml
md5: c3a73109be6c186b9d72e714bcedaddb
outs:
  - cache: true
    md5: 6836f797f3924fb46fcfd6b9f6aa6416.dir
    metric: false
    path: data/prepared
wdir: .
```

**Comment by jorgeorpinel** [comment]
Great. Yeah, Prettier forces it into the 2nd form you provided. Fixed all in b5cd910
